File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program174.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a java program which accept string from user and check whether it contains vowels in it or not.	
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
#include<stdio.h>

using namespace std;

class StringInput
{
	public:

		char Arr[30];

		void Accept()
		{
			cout<<"Enter the String: ";
			scanf("%[^'\n']s",Arr);
		}

};

class StringDemo:public StringInput
{
	public:

		bool ChkVowel()
		{
			char *str = Arr;
			bool bAns = false;
			int iCnt=0;

			while(*str!='\0')
			{
				if(*str=='a' || *str=='e' || *str=='i' || *str=='o' || *str=='u')
				{
					bAns=true;
					break;
				}
				str++;
			}

			return bAns;
		}
};

int main()
{
	bool bRet=false;
	StringDemo *sObj = new StringDemo();
	sObj->Accept();	
	bRet = sObj->ChkVowel();

	if(bRet==true)
	{
		cout<<"Vowel is present\n";
	}
	else
	{
		cout<<"Vowel is absent\n";
	}

	return 0;
}

/*
OUTPUT:

Enter the String: vishal
Vowel is present

*/

File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program174.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a java program which accept string from user and check whether it contains vowels in it or not.	
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class StringDemo
{
	public boolean ChkVowel(String str)
	{
		int iCnt=0;
		boolean bRet = false;
		while(iCnt!=str.length())
		{
			if(str.charAt(iCnt)=='a' || str.charAt(iCnt)=='e' || str.charAt(iCnt)=='i' || str.charAt(iCnt)=='o' || str.charAt(iCnt)=='u')
			{
				bRet = true;
				break;
			}
			iCnt++;
		}
		
		return bRet;
	}
}


class Program
{
	public static void main(String arg[])
	{
		boolean bRet = false;

		Scanner sObj = new Scanner(System.in);
		System.out.println("Input: ");
		String str = sObj.nextLine();
	
		StringDemo sobj = new StringDemo();
		bRet = sobj.ChkVowel(str);
	
		if(bRet==true)
		{
			System.out.println("Vowel is present");
		}
		else
		{	
			System.out.println("Vowel is absent");
		}	
	}
}

/*
OUTPUT:

Input:
Vishal
Vowel is present

*/
